lib/inputs/rhel/static.ts

```diff
@@ -0,0 +1,37 @@
+import {
+  ExtractAction,
+  ExtractedLayers,
+  RedHatRepo,
+  RedHatRepos,
+} from "../../extractor/types";
+import { streamToJson } from "../../stream-utils";
+
+export const getRedHatReposContentAction: ExtractAction = {
+  actionName: "redhat-content-manifests",
+  filePathMatches: isRedHatContentManifest,
+  callback: streamToJson,
+};
+
+export function getRedHatReposFromExtractedLayers(
+  extractedLayers: ExtractedLayers,
+): RedHatRepos {
+  const reposByLayer = {};
+  for (const filePath in extractedLayers) {
+    if (isRedHatContentManifest(filePath)) {
+      try {
+        const content = extractedLayers[filePath][
+          "redhat-content-manifests"
+        ] as any;
+        const layerIndex = content.metadata.image_layer_index;
+        reposByLayer[layerIndex] = content.content_sets as RedHatRepo;
+      } catch (err) {
+        throw new Error("Failed to extract repos information");
```

Comment:
So this will give up altogether if a single extracted layer is bad? Is this defintely what we want, rather than doing some kind of "this extracted layer is bad so I'll skip it, but I'll still handle the other ones that look legit" thing? (Genuine question, I don't know how we handle similar issues elsewhere :) )

Reply:
Great catch, I think it's way better to not break everything. Will change 💯